### DISCUSSION TITLE
Documentation clarification about transient_nonexcl_queues feature deprecation

### DISCUSSION THREAD

**Discussion by Tazmaniac** [rabbitmq/rabbitmq-server#12972] [Other] [answered]
### Is your feature request related to a problem? Please describe.

Per comments https://github.com/rabbitmq/rabbitmq-server/issues/12802#issuecomment-2548926329

We could deduce that after deprecation, only classic exclusive queues will be supported.
All other forms of classic queues without "durable:true" will not.

As the queue property is "durable:true" and not "transient:false" the affected queue types should be explicitly documented.

### Describe the solution you'd like

Describe the feature flag in https://www.rabbitmq.com/docs/feature-flags .
Explicitly state that after disabling this feature flag, only classic exclusive queue without "durable:true" will be supported all other non-exclusive classic queue requiring "durable:true".

**Comment by michaelklishin** [comment]
@Tazmaniac the docs are [open source](https://github.com/rabbitmq/rabbitmq-website), so anyone can edit them. However, there are several factually incorrect claims in what you expect us to document.

The term "transient" is [used in section dedicated to queue durability](https://www.rabbitmq.com/docs/queues#durability). In most client libraries `durable` is the only option to control the setting, so "transient" can only be expressed as "durable: false` but the term has been used in our docs for years.

Non-exclusive, transient/non-durable classic queues are still supported, and there are no plans to remove them in 4.1.0.
We cannot claim that "only classic exclusive queue without "durable:true" will be supported" since that is factually incorrect at the moment and will be incorrect till at least fall 2025, if not after that.

Finally, the combination of `durable` and `exclusive` flags does not make much sense, as the [Queues guide](https://www.rabbitmq.com/docs/queues) tries to explain by covering queue durability and exclusivity in separate sections and as separate "kinds of queues" (long-lived and shared vs. short-lived and client session-specific).

Feature flags cannot be disabled once enabled. You will have to enable all feature flags eventually, there is no way around it, if you plan on upgrading RabbitMQ.

**Comment by michaelklishin** [comment]
I will not document any queue properties in the feature flags guide. It makes no sense to me.

I will add a mention to the Queues guide that non-durable non-exclusive queues are deprecated and will eventually be removed.

**Comment by michaelklishin** [comment]
I guess [this list of deprecated features](https://www.rabbitmq.com/release-information/deprecated-features-list) can be expanded, too. But deprecated features are not feature flags. They have nothing in common besides the fact that you can enable both.

**Reply by michaelklishin** [reply, marked as answer]
Done https://github.com/rabbitmq/rabbitmq-website/commit/74e96c48ce4ac88e7ca6c9eb6472ffbd9fb3372c

**Reply by michaelklishin** [reply]
https://github.com/rabbitmq/rabbitmq-website/commit/6c01cc1dc4b8cfe76eef51f499a61993e3a7b030

**Reply by Tazmaniac** [reply]
Thank you, this is perfect !!
